src/components/NewDesign/Swap/Swap.tsx

```diff
@@ -0,0 +1,339 @@
+import React, { useEffect } from 'react'
+import { PublicKey } from '@solana/web3.js'
+import { BN } from '@project-serum/anchor'
+import { printBN, printBNtoBN } from '@consts/utils'
+import { Grid, Typography, Box, CardMedia } from '@material-ui/core'
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
+import { OutlinedButton } from '@components/NewDesign/OutlinedButton/OutlinedButton'
+import ExchangeAmountInput from '@components/NewDesign/Inputs/ExchangeAmountInput/ExchangeAmountInput'
+import useStyles from './style'
+import { Status } from '@reducers/solanaWallet'
+import SwapArrows from '@static/svg/swap-arrows.svg'
+import { PRICE_DECIMAL } from '@consts/static'
+
+export interface SwapToken {
+  balance: BN
+  decimal: number
+  symbol: string
+  assetAddress: PublicKey
+}
+
+export interface Pools {
+  tokenX: {
+    _bn: PublicKey
+  }
+  tokenY: {
+    _bn: PublicKey
+  }
+  sqrtPrice: {
+    v: BN
+  }
+}
+
+export interface ISwap {
+  walletStatus: Status
+  tokens: SwapToken[]
+  pools: Pools[]
+  onSwap: (fromToken: PublicKey, toToken: PublicKey, amount: BN) => void
+}
+export const Swap: React.FC<ISwap> = ({
+  walletStatus,
+  tokens,
+  pools,
+  onSwap
+}) => {
+  const classes = useStyles()
+  const [tokenFromIndex, setTokenFromIndex] = React.useState<number | null>(
+    tokens.length ? 0 : null
+  )
+  const [tokenToIndex, setTokenToIndex] = React.useState<number | null>(null)
+  const [amountFrom, setAmountFrom] = React.useState<string>('')
+  const [amountTo, setAmountTo] = React.useState<string>('')
+  const [collapsed, setCollapsed] = React.useState<boolean>(false)
+  const [swap, setSwap] = React.useState<boolean | null>(null)
+  const [priceProportion, setPriceProportion] = React.useState<BN>(new BN(1))
+  const [tokenY, setTokenY] = React.useState<SwapToken[] | null>(null)
+  // const firstUpdate = useRef(true)
+
+  const calculateSwapOutAmount = (assetIn: SwapToken, assetFor: SwapToken, amount: string) => {
+    // TODO: solution if 0 => change to 1
+    if (priceProportion.eqn(0)) {
+      setPriceProportion(new BN(1))
+    }
+
+    const amountOut: BN =
+      swap
+        ? printBNtoBN(amount, assetIn.decimal).mul(priceProportion)
+        : printBNtoBN(amount, assetIn.decimal).div(priceProportion)
+
+    return printBN(amountOut, assetFor.decimal)
+  }
+
+  useEffect(() => {
+    updateEstimatedAmount()
+
+    if ((tokenFromIndex !== null && tokenToIndex === null)) {
+      setAmountFrom('0.000000')
+    }
+    if (tokenFromIndex !== null) {
+      const tokensY = tokens.filter((token) => {
+        if (swap) {
+          return getSwapPoolIndex(token.assetAddress, tokens[tokenToIndex ?? 0].assetAddress) !== -1
+        } else {
+          return getSwapPoolIndex(token.assetAddress, tokens[tokenFromIndex].assetAddress) !== -1
+        }
+      })
+      setTokenY(tokensY)
+    }
+  }, [tokenToIndex, tokenFromIndex])
+
+  useEffect(() => {
+    if (tokenFromIndex === null || tokenToIndex === null) {
+
```

Comment:
why is this case empty?

Reply:
because i want to useEffect only if both indexes aren't null. I changed that using proper if statement for case and not use else statement